recipes/default.rb

```diff
@@ -20,7 +20,7 @@
 # Installs base package, creates log directory, and write out CDAP configuration
 include_recipe 'cdap::base'
 
-if node['cdap'].key?('cdap_env') && node['cdap']['cdap_env'].key?('log_dir')
+if cdap_property('log_dir', 'cdap_env')
```

Comment:
There's nothing in `cdap_env` by default and the recipe will NPE without this, as the resource name is derived from the attribute value. Resources cannot have `nil` as their name.